src/Microsoft.AspNet.Mvc.Core/Formatters/InputFormatter.cs

```diff
@@ -126,7 +132,12 @@ protected Encoding SelectCharacterEncoding(MediaTypeHeaderValue contentType)
             }
 
             // No supported encoding was found so there is no way for us to start reading.
-            throw new InvalidOperationException(Resources.FormatInputFormatterNoEncoding(GetType().FullName));
+            return null;
+        }
+
+        protected string GetNoEncodingMessage()
+        {
+            return Resources.FormatInputFormatterNoEncoding(GetType().FullName);
```

Comment:
So you've now seen the problem I solved here? Let's discuss alternatives tomorrow.